modules/caddyhttp/headers/caddyfile.go

```diff
@@ -27,11 +30,16 @@ func init() {
 	httpcaddyfile.RegisterHandlerDirective("request_header", parseReqHdrCaddyfile)
 }
 
+// ErrBadOperation is returned when trying to use a header operation in an inappropriate context.
+// For instance, when trying to use "?" operation (set default) with "header_down".
+var ErrBadOperation = errors.New("this header operation cannot be used here")
```

Comment:
Let's not export this; avoid cluttering the public API (exporting is a big commitment).